File: articles/grit-in-positive-psychology/go/grit_positive_psychology_score.go
package main

import "fmt"

func GritScore(perseveranceEffort float64, durableInterest float64) float64 {
	return 0.60*perseveranceEffort + 0.40*durableInterest
}

func main() {
	fmt.Printf("Synthetic grit score for positive psychology model: %.3f\n", GritScore(0.78, 0.66))
}

File: articles/grit-in-positive-psychology/rust/grit_positive_psychology_score.rs
fn grit_score(perseverance_effort: f64, durable_interest: f64) -> f64 {
    0.60 * perseverance_effort + 0.40 * durable_interest
}

fn main() {
    println!(
        "Synthetic grit score for positive psychology model: {:.3}",
        grit_score(0.78, 0.66)
    );
}
